feedfinder/internal/feedfinder/feedfinder.go

```diff
@@ -3,52 +3,66 @@ package feedfinder
 import (
 	"context"
 	"net/http"
+	"net/url"
 
 	"github.com/PuerkitoBio/goquery"
 	"github.com/sapuri/feed-finder/feedfinder/errors"
 )
 
+type Feed struct {
+	Title string
+	URL   *url.URL
+}
+
 type FeedFinder struct{}
 
 func NewFeedFinder() *FeedFinder {
 	return &FeedFinder{}
 }
 
-func (ff *FeedFinder) FindFeeds(ctx context.Context, siteURL string) (feedURLs []string, err error) {
+func (ff *FeedFinder) FindFeeds(ctx context.Context, siteURL string) ([]*Feed, error) {
 	// TODO: allow the HTTP option to be specified
 	client := http.DefaultClient
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, siteURL, nil)
 	if err != nil {
-		return
+		return nil, err
 	}
 	req.Header.Set("User-Agent", "Mozilla/5.0")
 
 	res, err := client.Do(req)
 	if err != nil {
-		return
+		return nil, err
 	}
 	defer func() {
 		_ = res.Body.Close()
 	}()
 
 	if res.StatusCode != http.StatusOK {
 		err = errors.NewHTTPError(res.StatusCode)
-		return
+		return nil, err
 	}
 
 	doc, err := goquery.NewDocumentFromReader(res.Body)
 	if err != nil {
-		return
+		return nil, err
 	}
 	links := doc.Find(`link[type="application/rss+xml"]`)
+	feeds := make([]*Feed, len(links.Nodes))
```

Comment:
[note to reviewers] Perf. update

Reply:
@KeisukeYamashita I thought it would be better to make the array variable length and continue if the url parsing fails, what do you think?

```suggestion
	feeds := make([]*Feed, 0, len(links.Nodes))
```

Reply:
Nice, LGTM 🎸